src/libs/actions/Task.ts

```diff
@@ -861,9 +859,7 @@ function deleteTask(report: OnyxEntry<OnyxTypes.Report>) {
             parentReport?.lastVisibleActionCreated ?? '',
             CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
         );
-        if (!isEmptyObject(optimisticParentReportData)) {
-            optimisticData.push(optimisticParentReportData);
-        }
+        optimisticParentReportData.forEach((parentReportData) => optimisticData.push(parentReportData));
```

Comment:
I notice 1 issue when deleting a task from a thread. Even though we already optimistically update the reply count to 0, the BE then will set it again back to 1 even though the task is deleted from the thread.

https://github.com/Expensify/App/assets/50919443/b704886f-5648-4eaf-9a3d-88edbd36822b